### PR TITLE
WV-2663 Modernize Palette Select

### PR DESCRIPTION
**Pull request by victorsh**
## Description

The PaletteSelect component is currently written as a React Class component. Today's best practices for React suggest we should be using functional components whenever possible. Using the modern best practices for React, convert the PaletteSelect component into a functional component.

1. Study the current behavior of the component so you know exactly what needs to be replicated.
2. Research the differences between class & functional components and how to replace lifecycle hooks with useEffect hooks.
3. Use the React docs to find the best practices for updating state, using hooks & props.
4. Use other components from the WV repo that are currently written as functional components for reference.
5. If there are obvious opportunities to refactor code to be more concise or easier to read do so.
6. After refactoring into a functional component, test the component to make sure the behavior is still the same.

Fixes # wv-2663

## How To Test

- `git fetch --all`
- `git checkout wv-2663-modernize-paletteselect`
- `npm ci`
- `npm run watch`
- Click on add layer
- Search for Aerosol Index
- Within the layer side-bar, for the Aerosol Index layer, select the layer options icon
- Selecting a different palette should update the map

@nasa-gibs/worldview

### CODE REVIEW

**Review by Tomcariello — COMMENTED** [review]
This looks good to me. Just the one minor nitpick.

File: web/js/components/layer/settings/palette.js [hunk 1]
```diff
@@ -119,15 +105,15 @@ class PaletteSelect extends React.Component {
           id={`wv-palette-radio-${id}`}
           type="radio"
           name="wv-palette-radio"
-          onClick={() => this.onChangePalette(id)}
+          onClick={() => onChangePalette(id)}
         />
         <label htmlFor={`wv-palette-radio-${id}`}>
           <img src={canvas.toDataURL('image/png')} />
```
[the remark this answers is not in the document]
Comment (victorsh) [reply web/js/components/layer/settings/palette.js]: Ok, I've updated this to store the function call in a variable.

**Review by Tomcariello — APPROVED** [review]

**Review by ryanweiler92 — CHANGES_REQUESTED** [review]
This seems pretty good to me except for one small thing. In the original class component we were importing `lodashIndexOf` from the `lodash` library and using it in the JSX when we mapped through `paletteOrder`. I saw that you replaced this with a traditional `indexOf` method. I tested this out and it seemed to work but there is a chance that the lodash version of indexOf was being used here to cover error handling for some edge cases that we may not be aware of through simple PR testing. I like to think that the developer that used this here had a good reason to use this instead of indexOf. I think to be safe we should put this back in instead of finding out the hard way that this was required for some edge cases. Here are some ways that `lodashIndexOf` differs from `indexOf`:

1. Handling of NaN values:

The indexOf method cannot find the index of NaN values in an array, as NaN !== NaN (i.e., NaN is not equal to any other value, including NaN itself). However, lodashIndexOf can find the index of NaN values in an array.

2. Optional fromIndex parameter:

The indexOf method has an optional fromIndex parameter, which specifies the starting index of the search. In contrast, lodashIndexOf also has a fromIndex parameter, but it is not optional, and the default value is 0.

3. Support for searching for a value within a range of indices:

lodashIndexOf can search for a value within a range of indices. It has two additional parameters: startIndex and endIndex, which allow you to specify the range of indices to search within. The indexOf method, on the other hand, only searches for a value starting from the fromIndex parameter.

**Comment by victorsh** [comment]
@ryanweiler92 Good call, I hadn't thought of that. My intent was to remove lodash entirely. My intent was to remove the need for another package but given the thorough explanation you provided above, it seems there may have been a good reason to use it instead of `indexOf`.

**Review by ryanweiler92 — APPROVED** [review]

**Pull request merged**